benchmark/definition/benchmarking.py

```diff
@@ -0,0 +1,42 @@
+"""Definition of the Benchmark class."""
+
+import time
+from collections.abc import Callable
+from uuid import UUID, uuid4
+
+from attrs import define, field
+from attrs.validators import instance_of
+from pandas import DataFrame
+
+from benchmark.result.result import Result
+
+
+@define
+class Benchmark:
```

Comment:
It would be cool if there was a `description` field which would just return the description of the actual `callable`. In this way, the `Benchmark` object itself would contain all possible relevant information.

Reply:
Great idea. Do you mean something like this to set it manually?
```suggestion
  description: str = field(validator=instance_of(str))
    """The description of the benchmark callable."""
```

Reply:
Was discussed in the meeting.

Reply:
As discussed to use the description of the callable and resolve it automatically, I would suggest the following. Had to restructure the callable docs in syntetic3.py as well.
```suggestion
    description: str = field(validator=instance_of(str), init=False)
    """The description of the benchmark callable. Will be 
    set as the docstring of the benchmark function automatically."""

    def __attrs_post_init__(self):
        """Set the description of the benchmark."""
        DESCRIPTION_NOT_SET = self.benchmark_function.__doc__ is None
        if DESCRIPTION_NOT_SET:
            raise ValueError("Description of the benchmark callable is not set.")
        self.description = self.benchmark_function.__doc__
```

Reply:
@fabianliebig, FYI: `__attrs_post_init__` is required only in very rare situations, mostly as a last resort. But this is not one of them and using it here is suboptimal in two ways:
* Instead, you could simply use the `default` decorator and instantiate it based on the given callable. That has the nice side-effect that it forces you to rethink the order of your attributes and pushes depend ones after independet ones.
* Even this is inelegant here since you'd simply duplicate information that way. If we need that piece, simply add it as `property` that reads the string from the callable attribute on demand.
* I'd say no validation is required here since our linters enforce us to add docstrings anyway and because a description is more like an optional thing. But if we wanted it, the validation would belong to the validator of the callable field. This also avoids the problem that a property should not raise an exception

Reply:
Thanks for pointing that out, I was not aware that this is a bad structure. Will change it as a property and add theoretical edge case handling for the linter: :)